doc/examples/filters/plot_deconvolution.py

```diff
@@ -2,54 +2,106 @@
 =====================
 Image Deconvolution
 =====================
-In this example, we deconvolve an image using Richardson-Lucy
-deconvolution algorithm ([1]_, [2]_).
-
-The algorithm is based on a PSF (Point Spread Function),
-where PSF is described as the impulse response of the
-optical system. The blurred image is sharpened through a number of
-iterations, which needs to be hand-tuned.
-
-.. [1] William Hadley Richardson, "Bayesian-Based Iterative
-       Method of Image Restoration",
-       J. Opt. Soc. Am. A 27, 1593-1607 (1972), DOI:10.1364/JOSA.62.000055
-
-.. [2] http://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
+In this example, Wiener, unsupervised Wiener, and Richardson-Lucy
+algorithms are used to deconvolve an image.
+Wiener and unsupervised Wiener filters
+--------------------------------------
+First, a noisy version of an image is deconvolved using Wiener
+and unsupervised Wiener algorithms. These algorithms are based on
+linear models that cannot restore sharp edges as much as non-linear
+methods (like TV restoration), but are faster.
+The Wiener deconvolution algorithm is an inverse filter based on an
+impulse response function (also called point-spread function, or PSF), a
+filter to penalize the restored image frequencies (regularization), and
+a parameter which tunes the balance between the data and the
+regularization.
+The unsupervised Wiener algorithm has self-tuned regularization
```

Comment:
Same here: the functional is not explicit, and the regularization impact is unclear.
Does mentioning Gibbs sampling clarify things here?